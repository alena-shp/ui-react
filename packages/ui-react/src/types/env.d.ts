/// <reference types="vite/client" />

/* Custom ENV
interface ImportMetaEnv {
    readonly MY_ENV: string
}
*/

interface ImportMeta {
    readonly env: ImportMetaEnv
}
