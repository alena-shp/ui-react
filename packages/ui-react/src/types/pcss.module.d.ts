declare module '*.module.pcss' {
    const classes: Record<string, string>;

    export default classes;
}
