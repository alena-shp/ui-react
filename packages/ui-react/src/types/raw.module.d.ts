declare module '*?raw' {
    const text: string;

    export default text;
}
