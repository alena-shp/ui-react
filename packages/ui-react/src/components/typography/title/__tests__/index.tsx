import React from 'react';
import * as renderer from 'react-test-renderer';

import { Title, typographyTitleTags } from '..';
import { typographyColor } from '../../types';

describe('Title', () => {
    typographyTitleTags.map((tag) => {
        return typographyColor.map((color) => {
            test(`${tag}: ${color}`, () => {
                const tree = renderer.create(
                    <Title
                        key={`${tag}: ${color}`}
                        tagName={tag}
                        presetColor={color}
                    >
                        {`${tag}: ${color}`}
                    </Title>
                ).toJSON();

                expect(tree).toMatchSnapshot();
            });
        });
    });
});
