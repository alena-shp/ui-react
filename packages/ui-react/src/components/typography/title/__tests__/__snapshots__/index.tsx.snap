// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title h1: basic 1`] = `
<h1
  className="typography typography_size-h1 typography_color-basic"
>
  h1: basic
</h1>
`;

exports[`Title h1: brand 1`] = `
<h1
  className="typography typography_size-h1 typography_color-brand"
>
  h1: brand
</h1>
`;

exports[`Title h1: brand-1 1`] = `
<h1
  className="typography typography_size-h1 typography_color-brand-1"
>
  h1: brand-1
</h1>
`;

exports[`Title h1: disabled 1`] = `
<h1
  className="typography typography_size-h1 typography_color-disabled"
>
  h1: disabled
</h1>
`;

exports[`Title h1: error 1`] = `
<h1
  className="typography typography_size-h1 typography_color-error"
>
  h1: error
</h1>
`;

exports[`Title h1: headers 1`] = `
<h1
  className="typography typography_size-h1 typography_color-headers"
>
  h1: headers
</h1>
`;

exports[`Title h1: inverse 1`] = `
<h1
  className="typography typography_size-h1 typography_color-inverse"
>
  h1: inverse
</h1>
`;

exports[`Title h1: link 1`] = `
<h1
  className="typography typography_size-h1 typography_color-link"
>
  h1: link
</h1>
`;

exports[`Title h1: secondary 1`] = `
<h1
  className="typography typography_size-h1 typography_color-secondary"
>
  h1: secondary
</h1>
`;

exports[`Title h1: success 1`] = `
<h1
  className="typography typography_size-h1 typography_color-success"
>
  h1: success
</h1>
`;

exports[`Title h1: warning 1`] = `
<h1
  className="typography typography_size-h1 typography_color-warning"
>
  h1: warning
</h1>
`;

exports[`Title h2: basic 1`] = `
<h2
  className="typography typography_size-h2 typography_color-basic"
>
  h2: basic
</h2>
`;

exports[`Title h2: brand 1`] = `
<h2
  className="typography typography_size-h2 typography_color-brand"
>
  h2: brand
</h2>
`;

exports[`Title h2: brand-1 1`] = `
<h2
  className="typography typography_size-h2 typography_color-brand-1"
>
  h2: brand-1
</h2>
`;

exports[`Title h2: disabled 1`] = `
<h2
  className="typography typography_size-h2 typography_color-disabled"
>
  h2: disabled
</h2>
`;

exports[`Title h2: error 1`] = `
<h2
  className="typography typography_size-h2 typography_color-error"
>
  h2: error
</h2>
`;

exports[`Title h2: headers 1`] = `
<h2
  className="typography typography_size-h2 typography_color-headers"
>
  h2: headers
</h2>
`;

exports[`Title h2: inverse 1`] = `
<h2
  className="typography typography_size-h2 typography_color-inverse"
>
  h2: inverse
</h2>
`;

exports[`Title h2: link 1`] = `
<h2
  className="typography typography_size-h2 typography_color-link"
>
  h2: link
</h2>
`;

exports[`Title h2: secondary 1`] = `
<h2
  className="typography typography_size-h2 typography_color-secondary"
>
  h2: secondary
</h2>
`;

exports[`Title h2: success 1`] = `
<h2
  className="typography typography_size-h2 typography_color-success"
>
  h2: success
</h2>
`;

exports[`Title h2: warning 1`] = `
<h2
  className="typography typography_size-h2 typography_color-warning"
>
  h2: warning
</h2>
`;

exports[`Title h3: basic 1`] = `
<h3
  className="typography typography_size-h3 typography_color-basic"
>
  h3: basic
</h3>
`;

exports[`Title h3: brand 1`] = `
<h3
  className="typography typography_size-h3 typography_color-brand"
>
  h3: brand
</h3>
`;

exports[`Title h3: brand-1 1`] = `
<h3
  className="typography typography_size-h3 typography_color-brand-1"
>
  h3: brand-1
</h3>
`;

exports[`Title h3: disabled 1`] = `
<h3
  className="typography typography_size-h3 typography_color-disabled"
>
  h3: disabled
</h3>
`;

exports[`Title h3: error 1`] = `
<h3
  className="typography typography_size-h3 typography_color-error"
>
  h3: error
</h3>
`;

exports[`Title h3: headers 1`] = `
<h3
  className="typography typography_size-h3 typography_color-headers"
>
  h3: headers
</h3>
`;

exports[`Title h3: inverse 1`] = `
<h3
  className="typography typography_size-h3 typography_color-inverse"
>
  h3: inverse
</h3>
`;

exports[`Title h3: link 1`] = `
<h3
  className="typography typography_size-h3 typography_color-link"
>
  h3: link
</h3>
`;

exports[`Title h3: secondary 1`] = `
<h3
  className="typography typography_size-h3 typography_color-secondary"
>
  h3: secondary
</h3>
`;

exports[`Title h3: success 1`] = `
<h3
  className="typography typography_size-h3 typography_color-success"
>
  h3: success
</h3>
`;

exports[`Title h3: warning 1`] = `
<h3
  className="typography typography_size-h3 typography_color-warning"
>
  h3: warning
</h3>
`;

exports[`Title h4: basic 1`] = `
<h4
  className="typography typography_size-h4 typography_color-basic"
>
  h4: basic
</h4>
`;

exports[`Title h4: brand 1`] = `
<h4
  className="typography typography_size-h4 typography_color-brand"
>
  h4: brand
</h4>
`;

exports[`Title h4: brand-1 1`] = `
<h4
  className="typography typography_size-h4 typography_color-brand-1"
>
  h4: brand-1
</h4>
`;

exports[`Title h4: disabled 1`] = `
<h4
  className="typography typography_size-h4 typography_color-disabled"
>
  h4: disabled
</h4>
`;

exports[`Title h4: error 1`] = `
<h4
  className="typography typography_size-h4 typography_color-error"
>
  h4: error
</h4>
`;

exports[`Title h4: headers 1`] = `
<h4
  className="typography typography_size-h4 typography_color-headers"
>
  h4: headers
</h4>
`;

exports[`Title h4: inverse 1`] = `
<h4
  className="typography typography_size-h4 typography_color-inverse"
>
  h4: inverse
</h4>
`;

exports[`Title h4: link 1`] = `
<h4
  className="typography typography_size-h4 typography_color-link"
>
  h4: link
</h4>
`;

exports[`Title h4: secondary 1`] = `
<h4
  className="typography typography_size-h4 typography_color-secondary"
>
  h4: secondary
</h4>
`;

exports[`Title h4: success 1`] = `
<h4
  className="typography typography_size-h4 typography_color-success"
>
  h4: success
</h4>
`;

exports[`Title h4: warning 1`] = `
<h4
  className="typography typography_size-h4 typography_color-warning"
>
  h4: warning
</h4>
`;
